packages/payment/src/services/payment-module.ts

```diff
@@ -201,44 +201,67 @@ export default class PaymentModuleService<
     )
   }
 
-  @InjectTransactionManager("baseRepository_")
+  @InjectManager("baseRepository_")
   async createPaymentSession(
     paymentCollectionId: string,
-    data: CreatePaymentSessionDTO,
+    input: CreatePaymentSessionDTO,
     @MedusaContext() sharedContext?: Context
   ): Promise<PaymentSessionDTO> {
-    const created = await this.paymentSessionService_.create(
-      {
-        provider_id: data.provider_id,
-        amount: data.providerContext.amount,
-        currency_code: data.providerContext.currency_code,
-        payment_collection: paymentCollectionId,
-      },
-      sharedContext
-    )
+    let paymentSession: PaymentSession
 
     try {
-      const sessionData = await this.paymentProviderService_.createSession(
-        data.provider_id,
-        {
-          ...data.providerContext,
-          resource_id: created.id,
-        }
-      )
+      const providerSessionSession =
```

Comment:
@adrien2p, wdyt?

Reply:
sounds good to me 💪

Reply:
the drawback is that we lose the built-in ability of the workflow to retry calling the provider, and in the future when we handle errors to compensate, we'll miss it here

Reply:
If you had to choose, what approach would you go with?

Reply:
The only problem with retry mechanism handled in that place could timeout the requests, probably the retry should happen from the client in that context

Reply:
I don't have a strong opinion, I just feel that by using the workflow we could benefit from future improvements and handling things using config, like the retries.
But that can be applied in the future as well, so I'm good with that too.
If we had more than 1 external call in the same operation then I would say the workflow would be a better choice.

Reply:
yeah, we can discuss that later but this is an external call, it is common to have built in retries when doing network calls as issues are somewhat common to happen.

Reply:
Yeah I do agree, my question is where this should happen, because if the request end up timing out while it is still retrying then you are loosing the context no? we can go with server retries I don't have a strong opinion I just wanted to raise this though ahah :)

Reply:
Alright, I will keep it as is, and we can revisit in the future.